zkevm-circuits/src/state_circuit/constraint_builder.rs

```diff
@@ -226,10 +230,98 @@ impl<F: Field> ConstraintBuilder<F> {
         // TODO: Missing constraints
     }
 
+    fn build_tx_log_constraints(&mut self, q: &Queries<F>) {
+        self.require_equal(
+            "is_write is always true for TxLog",
+            q.is_write.clone(),
+            1.expr(),
+        );
+        self.require_in_set(
+            "field_tag in TxLogFieldTag range",
+            q.field_tag(),
+            set::<F, TxLogFieldTag>(),
+        );
+        // reset log_id when tx_id increases
+        self.condition(q.id_change() * q.is_tag_unchanged.clone(), |cb| {
+            cb.require_zero(
+                "reset log_id to one when tx_id increases",
+                q.tx_log_id(false) - 1.expr(),
+            );
+            // constrain first field_tag is Address when tx id increases
+            cb.require_equal(
+                "first field_tag is Address when tx changes",
+                q.field_tag_matches(TxLogFieldTag::Address),
+                1.expr(),
+            );
+            // tx_id increases by 1
+            cb.require_equal("tx_id increases by 1", q.id_change(), 1.expr());
+        });
+
+        self.condition(q.field_tag_matches(TxLogFieldTag::Address), |cb| {
+            cb.require_zero("index is zero for address ", q.tx_log_index(false))
```

Comment:
In the spec, this constraint is enforced when `row.tag() == row_prev.tag()` and `tx_id == pre_tx_id`.  Maybe update the spec?